### DISCUSSION TITLE
How to specify `imagePullPolicy` for RabbitMQ image?

### DISCUSSION THREAD

**Discussion by tkoft** [rabbitmq/cluster-operator#1566] [Q&A]
This may just be my knowledge gap with kubernetes, but how can I specify the `imagePullPolicy` for RabbitMQ pods in my cluster operator definition? The [StatefulSetSpec for overriding the default StatefulSet configuration](https://github.com/rabbitmq/cluster-operator/blob/main/docs/api/rabbitmq.com.ref.asciidoc#statefulsetspec) doesn't include it in its API.

Higher level goal is to use one of the major-version tags of the rabbitmq docker image (e.g. `3.12-management`), and have it be re-pulled when pods are redeployed (if the image digest has changed, which `imagePullPolicy: Always` should check) so we don't have to manually update it with each minor version.

**Comment by tkoft** [comment]
Ah I did find it under `RabbitMqCluster.spec.override.statefulSet.spec.template.spec.containers.imagePullPolicy`. 

Regarding `template`:
> `template` is the object that describes the pod that will be created if insufficient replicas are detected. Each pod stamped out by the StatefulSet will fulfill this Template, but have a unique identity from the rest of the StatefulSet.

Does this mean updating the template will not reconcile it into existing pods?

**Reply by Zerpet** [reply]
Those docs are a copy paste from Kubernetes Statefulset documentation. Pods will be updated if there are changes to the `template` field.
